### PR TITLE
NET-5392 Replace dev dependencies w/ latest release tags

### PR DESCRIPTION
**Pull request by nathancoleman**
**Changes proposed in this PR:**
Some dependencies were previously pinned to utilize unreleased features of the Consul SDK.
This replaces those with the tagged releases containing those features

**How I've tested this PR:**

**How I expect reviewers to test this PR:**

**Checklist:**
- [ ] Tests added
- [ ] [CHANGELOG entry added](https://github.com/hashicorp/consul-k8s/blob/main/CONTRIBUTING.md#adding-a-changelog-entry)

### CODE REVIEW

**Review by DanStough — APPROVED** [review]

File: control-plane/api-gateway/common/diff.go [hunk 1]
```diff
@@ -274,8 +274,10 @@ func (e entryComparator) urlRewritesEqual(a, b api.URLRewrite) bool {
 }
 
 func (e entryComparator) retryFiltersEqual(a, b api.RetryFilter) bool {
-	return BothNilOrEqual(a.NumRetries, b.NumRetries) && BothNilOrEqual(a.RetryOnConnectFailure, b.RetryOnConnectFailure) &&
-		slices.Equal(a.RetryOn, b.RetryOn) && slices.Equal(a.RetryOnStatusCodes, b.RetryOnStatusCodes)
+	return a.NumRetries == b.NumRetries &&
```
Comment (jm96441n) [inline control-plane/api-gateway/common/diff.go:277]: `NumRetries` here is still a pointer so we need the `BothNilOrEqual` right?

Comment (nathancoleman) [reply control-plane/api-gateway/common/diff.go:277]: It's just a `uint32` now

https://github.com/hashicorp/consul/blob/api/v1.26.1/api/config_entry_routes.go#L225

**Review by jm96441n — APPROVED** [review]
LGTM

File: control-plane/api/mesh/v2beta1/proxy_configuration_types_test.go [hunk 2]
```diff
@@ -109,20 +108,6 @@ func TestProxyConfiguration_MatchesConsul(t *testing.T) {
 							JsonFormat:          "jsonFormat",
 							TextFormat:          "text format.",
 						},
-						EnvoyExtensions: []*pbmesh.EnvoyExtension{
```
Comment (nathancoleman) [inline control-plane/api/mesh/v2beta1/proxy_configuration_types_test.go:112]: This field was intentionally removed in https://github.com/hashicorp/consul/pull/19186.
I've confirmed with @ishustava that it's safe to remove this code from the test.

**Review by NicoletaPopoviciu — APPROVED** [review]

**Pull request merged**